Fetch/Fetch.php

```diff
@@ -461,4 +467,41 @@ private function checkDeep($params)
 
         return is_null($param) ? $this->getConfigBool('deep') : bool($param);
     }
+
+    private function addChildPagesToPage(\Statamic\Data\Pages\Page $page): array
+    {
+        $data = $this->getPageData($page);
+
+        $depth = $this->depth > 0 ? $this->depth : null;
+        $children = collect(Content::tree($data['uri'], $depth, false, false, null, $this->locale));
+        $children = $children->map(function ($item) {
+            return $this->processNestedPage($item);
+        })->all();
+
+        $data['children'] = $children;
+
+        return $data;
+    }
+
+    private function processNestedPage(array $item): array
+    {
+        $page = $this->getPageData($item['page']);
+
+        if ($item['children']) {
+            $page['children'] = collect($item['children'])->map(function ($item) {
+                return $this->processNestedPage($item);
+            })->all();
+        }
+
+        return $page;
+    }
+
+    private function getPageData(\Statamic\Data\Pages\Page $page): array
+    {
+        if ($this->deep) {
+            $page->supplementTaxonomies();
```

Comment:
I am not sure If I need to use `dataForLocale` 😕 
Or the `goDeep`  method? 🤷‍♂️

Reply:
I'll take a look at this during review 👍